src/include/txn_inline.h

```diff
@@ -662,7 +662,7 @@ __wt_txn_pinned_stable_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinne
      */
     WT_ACQUIRE_READ(pinned_stable_ts, txn_global->stable_timestamp);
 
-    if (!__wt_conn_is_disagg(session)) {
+    if (!F_ISSET(S2C(session), WT_CONN_PRECISE_CHECKPOINT)) {
```

Comment:
Perf nit: since this is the second use of `S2C` in this function, probably good to have a `conn` variable to cache the connection.